### PR TITLE
pkg/p2p(cdc): fix minor bugs and add basic integration tests

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- We need some test cases for the `MessageServer` and `MessageClient` to work together.
- We need to resolve some minor problems.

### What is changed and how it works?
- Added `server_client_integration_test.go`.
- Fixed a deadlock bug in `MessageClient`.
- Do not GC `ackManager` for now. Better GC mechanism will be added in a subsequent PR.
- Fixed unstable unit test `TestMessageClientBasics`.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: pkg/p2p/server_client_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+	cancel = func() {
+		grpcServer.Stop()
+		wg.Wait()
+	}
+	return
+}
+
+func runP2PIntegrationTest(ctx context.Context, t *testing.T, size int, numTopics int) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
```
Comment (amyangfei) [inline pkg/p2p/server_client_integration_test.go:81]: L163 already calls `cancel()`

Comment (liuzix) [reply pkg/p2p/server_client_integration_test.go:81]: Added the deferred cancel here in case there is a panic mid-way.

**Review by amyangfei — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**